src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/LegacyPbaAccountServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.reform.professionalapi.service.impl;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.stereotype.Service;
+
+import uk.gov.hmcts.reform.professionalapi.domain.PaymentAccount;
+import uk.gov.hmcts.reform.professionalapi.domain.ProfessionalUser;
+import uk.gov.hmcts.reform.professionalapi.domain.UserAccountMap;
+import uk.gov.hmcts.reform.professionalapi.service.LegacyPbaAccountService;
+
+@Service
+@Slf4j
+public class LegacyPbaAccountServiceImpl implements LegacyPbaAccountService {
+
+
+
+    public List<String> findLegacyPbaAccountByUserEmail(ProfessionalUser professionalUser) {
+
+        List<String> pbaNumbers = null;
+
+        List<PaymentAccount>  paymentAccountsFromEntity = new ArrayList<>();
+
+        if (!professionalUser.getOrganisation().getPaymentAccounts().isEmpty()) {
+
+            List<PaymentAccount>  userMapPaymentAccount = getPaymentAccountsFromUserAccountMap(professionalUser.getUserAccountMap());
+
+            pbaNumbers = getPbaNumbersFromPaymentAccount(userMapPaymentAccount, professionalUser.getOrganisation().getPaymentAccounts());
+
+        }
+
+        return pbaNumbers;
+    }
+
+    private List<String> getPbaNumbersFromPaymentAccount(List<PaymentAccount> userMapPaymentAccount, List<PaymentAccount> paymentAccountsEntity) {
+
+        List<String> paymentAccountPbaNumbers = new ArrayList<>();
+
+        if (!paymentAccountsEntity.isEmpty()) {
+
+            paymentAccountsEntity.forEach(paymentAccount -> {
+                for (PaymentAccount usrMapAccount : userMapPaymentAccount) {

```

Comment:
The for loop could be removed with a bit of refactoring to more completely use Java 8 streams/lambdas, but ok for now (this can be addressed in a follow-up refactoring jira).